nixos/tests/installer.nix

```diff
@@ -332,6 +334,40 @@ in {
         '';
     };
 
+  # zfs on / with swap
+  zfs_root = makeInstallerTest "zfs-root"
+    {
+      extraInstallerConfig = {
+        boot.supportedFilesystems = [ "zfs" ];
+      };
+
+      extraConfig = ''
+        boot.supportedFilesystems = [ "zfs" ];
+        boot.loader.grub.zfsSupport = true;
+        boot.zfs.devNodes = "/dev/disk/by-uuid/";
```

Comment:
qemu vda disks don't have an ID, and don't show up in the default `/dev/disk/by-id`.